controller/controller.go

```diff
@@ -113,59 +117,76 @@ func (c *PokemonController) PostPokemon(w http.ResponseWriter, r *http.Request)
 	}
 }
 
+/// in this method recives the parameters from the request , get the vquery parameters type , items , items_per_worker  validate the corect values
+/// then it rerurs an error if it occurs or return a list op pokemons
 func (c *PokemonController) GetPokemonsConcurrently(w http.ResponseWriter, r *http.Request) {
 	types := r.URL.Query().Get("type")
 	items := r.URL.Query().Get("items")
 	itemsPW := r.URL.Query().Get("items_per_workers")
 	defer r.Body.Close()
+	w.Header().Add("Content-Type", "application/json")
 
-	itemsInt, errItems := strconv.Atoi(items)
-	items_per_worker, errWorker := strconv.Atoi(itemsPW)
-
-	if errItems != nil || errWorker != nil || items_per_worker == 0 || itemsInt == 0 || itemsInt%items_per_worker != 0 || types != "even" && types != "odd" || itemsInt < items_per_worker {
-
-		err := ""
-
-		if types != "even" && types != "odd" {
-			err = "Type should be even or odd"
-
-		} else if errWorker != nil || items_per_worker == 0 {
-			err = "Items_per_workers should be a valid number more than 0"
-
-		} else if errItems != nil || items_per_worker == 0 {
+	itemsInt, err := strconv.Atoi(items)
+	if err != nil && itemsInt == 0 {
+		err = errors.Wrap(err, "items should be an integer")
+		w.WriteHeader(http.StatusBadRequest)
+		encoderErr := json.NewEncoder(w).Encode(map[string]string{"error": "items should be an integer"})
+		if err != nil {
+			err = errors.Wrap(err, encoderErr.Error())
+		}
+		log.Println(err)
 
-			err = "Items should be a valid number more than 0"
+		return
+	}
 
-		} else if itemsInt%items_per_worker != 0 {
-			err = "items_per_workers should be a multiple of items"
-		} else if itemsInt < items_per_worker {
-			err = "items should be greather than items_per_workers"
+	items_per_worker, errWorker := strconv.Atoi(itemsPW)
+	if errWorker != nil && items_per_worker == 0 {
```

Comment:
Should be `err`. I assume `errWorker` is a specific error for when calling the worker usecase, not when checking if `items_per_worker` (which should be `itemsPerWorker` - no snake case, remember) is an int.